src/EditorFeatures/Core/Implementation/CodeCleanup/CodeCleanupService.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.OrganizeImports;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.CodeCleanup
+{
+    [Export(typeof(ICodeCleanupService))]
+    internal class CodeCleanupService : ICodeCleanupService
+    {
+        /// <summary>
+        /// Maps format document code cleanup options to DiagnosticId[]
+        /// </summary>
+        private static ImmutableDictionary<PerLanguageOption<bool>, string[]> _dictionary = GetCodeCleanupOptionMapping();
+
+        private readonly ICodeFixService _codeFixService;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CodeCleanupService(
+            ICodeFixService codeFixService)
+        {
+            _codeFixService = codeFixService;
+        }
+        public static IDictionary<PerLanguageOption<bool>, string[]> Dictionary
+        {
+            get
+            {
+                return _dictionary;
+            }
+        }
+
+        private static ImmutableDictionary<PerLanguageOption<bool>, string[]> GetCodeCleanupOptionMapping()
+        {
+            var dictionary = new Dictionary<PerLanguageOption<bool>, string[]>();
+            dictionary.Add(FeatureOnOffOptions.FixImplicitExplicitType,
```

Comment:
1. it looks like this could all be simplified with a dictionary initializer.
2. it feels super weird to have this list hardcoded.... is it not possible for these individual features to implement some interface that could be introspected to learn this data so that this service doesn't need to hardcode this all in?